chore: add EVM extensions article (#13)
evmos/academy: docs/articles/advanced/evm-extensions-stk-distr.md

  - [Authorization.sol](https://github.com/evmos/extensions/blob/main/precompiles/common/Authorization.sol)
  - [GenericAuthorization.sol](https://github.com/evmos/extensions/blob/main/precompiles/common/GenericAuthorization.sol)
  - [SimpleStaker.sol](https://github.com/evmos/extensions/blob/main/examples/simple-staker/contracts/SimpleStaker.sol)
- Fix the imports. If you're pasting all these files in the `contracts` directory,
  make sure to update the imports, for example, in `Staking.sol`:

  ```solidity
    import "../common/Authorization.sol" as authorization;
    import "../common/Types.sol";
  ```

  Replace with:

  ```solidity
    import "./Authorization.sol" as authorization;
    import "./Types.sol";
  ```

- Compile the contract with Remix

## Step 2: Connect MetaMask to the Evmos testnet

Follow [the corresponding guide on the Evmos docs](https://docs.evmos.org/use/connect-your-wallet/metamask) to achieve this.

## Step 3: Get some funds

If you don't have tEVMOS on your wallet,
get some tokens from the [Evmos Testnet Faucet](https://faucet.evmos.dev/).

## Step 4: Deploy the `SimpleStaker.sol` contract

> - Go to ***Deploy & Run Transactions*** option on Remix side bar

Reviewer: ```suggestion
- Go to ***Deploy & Run Transactions*** option on Remix's side bar
```

> - Configure Remix Environment to use ***Injected Provider - MetaMask***

Reviewer: ```suggestion
- Configure ***Injected Provider - MetaMask*** as the used environment
```

- Select the account with funds from the previous step
- Make sure the `SimpleStaker` compiled contract is selected in the ***CONTRACT*** field

> - Deploy the smart contract by pressing the ***Deploy*** button & approving the transaction on Metamask.

Reviewer: nit: I would either use dots at the end of the points everywhere or nowhere
```suggestion
- Deploy the smart contract by pressing the ***Deploy*** button & approving the transaction on Metamask
```

> ![prepare deployment](/img/remix_prepare_deploy.png)

Reviewer: I think it would look nicer if the images were centered -- I think you can use a simple HTML div element for that? not sure if this works in docusaurus but worked before in the old docs if I'm not mistaken?

Upon a successful deployment,
you should see the contract with its exposed methods being listed under ***Deployed Contracts***.

![contract deployed](/img/remix_deployed.png)

## Step 5: Interact with the deployed contract

### Get current validators

To use the `SimpleStaker` contract, we need a validator's address.
Check the current validators using the following command:

```bash
evmosd q staking validators --node https://tendermint.bd.evmos.dev:26657
```

Copy the `operator_address` of the validator you want to use.
Use this as the `_validatorAddr` in the `SimpleStaker` functions.

### Grant Approvals

Before performing any transaction,
make sure to set the corresponding approvals.
To do so, call the `approveRequiredMethods` function.
This function will grant permissions to the smart contract
to use the `MsgDelegate` and `MsgWithdrawDelegatorWithdraw`
on behalf of the caller.

### Stake some tokens

Call the `stakeTokens` function using the validator address obtained previously
and the desired amount to stake.

Once the transaction is processed successfully,
you can check your current delegations calling the `getDelegation` function.

> Additionally, you can check the current delegations using the evmos binary with the following command:
> 
> ```bash
> BECH32ADDR=$(evmosd debug addr <YOUR_METAMASK_ADDR> | grep "Bech32 Acc" | cut -c 13-56)
> evmosd q staking delegations $BECH32ADDR --node https://tendermint.bd.evmos.dev:26657
> ```

Reviewer: ```suggestion
Additionally, you can check the current delegations using the [Swagger API page](https://api.evmos.dev/#/Query/DelegatorDelegations) where you use your Bech32-formatted address as the delegator.
```
